htdocs/core/lib/functions.lib.php

```diff
@@ -8064,7 +8064,13 @@ function getCommonSubstitutionArray($outputlangs, $onlykey = 0, $exclude = null,
 			if (is_object($object) && $object->element == 'project') {
 				$substitutionarray['__PROJECT_NAME__'] = $object->title;
 			}
-
+			if (is_object($object) && $object->element == 'propal' && !empty($object->fk_project)) {
```

Comment:
We should make fetch and database acces into this function.
If the property about proejct should be filled if property $object->projet is correctly filled.
The problem seems to be in the code that is calling getCommonSubstitutionArray()
The method   $object->fetch_project() is probably not called before calling the getCommonSubstitutionArray()

Reply:
Thank you for the advice.
Changes on functions.lib are cancelled, and I added fetch_project() in propal.class.
This solves the issue in a better way